PolyEngine/Core/Src/String.cpp

```diff
@@ -0,0 +1,207 @@
+#include"CorePCH.hpp"
+
+#include"String.hpp"
+
+using namespace Poly;
+
+size_t Poly::StrLen(const char* str) {
+	size_t len = 0;
+	while (str[len] != 0)
+		++len;
+	return len;
+}
+
+
+String::String(const char* data) {
+	size_t length = StrLen(data);
+	Data.Resize(length + 1);
+	std::memcpy(Data.GetData(), data, sizeof(char) * length);
+	Data[length] = 0;
+}
+
+String::String(const String& rhs) {
+	*this = rhs;
+}
+
+String::String(String&& rhs) {
+	*this = String(std::move(rhs));
+}
+
+
+String* String::From(int var) {
+	return new String(std::to_string(var).c_str());
+}
+
+String* String::From(float var) {
+	return new String(std::to_string(var).c_str());
+}
+
+String* String::From(double var) {
+	return new String(std::to_string(var).c_str());
+}
+
+String* String::From(char var) {
+	return From(std::string(1, var));
+}
+
+String* String::From(const char* var) {
+	return new String(var);
+}
+
+String* String::From(const std::string& var) {
+	return new String(var.c_str());
+}
+
+
+bool String::Contains(const String& var) const {
+	//TODO Dynarray.Contains(Dynarray<T> rhs)
+	return false;
+}
+
+String* String::ToLower() const {
+	Dynarray<char> d = Dynarray<char>(this->Data.GetSize());
```

Comment:
This method results in double allocation( first dynarray,m second new string), create empty string, resize its dynarray, and fill it. Remove dynarray `d` alltogether.